packages/react-router/__tests__/data-memory-router-test.tsx

```diff
@@ -2051,32 +2058,41 @@ describe("<DataMemoryRouter>", () => {
         return (
           <>
             <MemoryNavigate to="/bar">Link to Bar</MemoryNavigate>
-            <p>{navigation.state}</p>
-            <Outlet />
+            <MemoryNavigate to="/baz">Link to Baz</MemoryNavigate>
+            <div id="content">
+              <p>{navigation.state}</p>
+              <Outlet />
+            </div>
           </>
         );
       }
 
-      function Foo() {
-        return <h1>Foo</h1>;
-      }
       function Bar() {
         let data = useLoaderData();
         return (
           <>
             <p>{data.critical}</p>
             <React.Suspense fallback={<LazyFallback />}>
-              <Await
-                resolve={data.lazy}
-                errorElement={hasAwaitErrorElement ? <LazyError /> : null}
-              >
-                {useRenderProp ? (value) => <p>{value}</p> : <LazyData />}
-              </Await>
+              <AwaitCounter data={data} />
             </React.Suspense>
           </>
         );
       }
 
+      function AwaitCounter({ data }) {
+        awaitRenderCount++;
```

Comment:
Leveraging this to assert that we aren't continually re-rendering during the transition that triggered the abort